### PR TITLE
feat(acp): consolidate ACP improvements

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

Consolidates ACP improvements from 6 separate PRs into one clean changeset:

- **agentInfo in InitializeResponse** (#1442, #1451): Return `Implementation(name="gptme", title="gptme ACP Agent", version=...)` so ACP clients can identify the agent
- **Multi-line command output formatting** (#1446): Wrap multi-line `/help`, `/tools` output in code blocks for proper rendering in ACP panels
- **Code cleanup** (#1447): Simplify permission policy caching with `setdefault()`, remove redundant f-string formatting, expand truncated session IDs in logs (8 -> 16 chars)
- **load_session returns modes/models** (#1448): `_build_load_session_response()` helper ensures loaded sessions get the same `modes`/`models` state as new sessions
- **Session resume on reconnect** (#1443): Derive deterministic session IDs from workspace path (`acp-<sha256[:8]>`) so editor restarts automatically resume the previous conversation. Persists workspace metadata via `ChatConfig.save()` for session discovery
- **Registry log update fix**: When a session exists in the registry with `log=None` (e.g. partial init), correctly update the log reference on resume

Supersedes: #1442, #1443, #1446, #1447, #1448, #1451

**Note**: GLM/OpenRouter changes (CI continue-on-error, max_tokens cap) removed per review feedback — this PR now focuses purely on ACP improvements.

## Test plan

- [x] All ACP agent tests pass (including new tests for agentInfo, load_session, session resume, registry log update)
- [ ] CI passes
- [ ] Manual test with Zed: verify agentInfo shows in client, session resumes after restart

## Files changed

| File | Changes |
|------|---------|
| `gptme/acp/agent.py` | ACP improvements consolidated |
| `tests/test_acp_agent.py` | New tests (agentInfo, load_session modes, session resume, cwd hashing, registry log) |

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: gptme/acp/agent.py [hunk 1]
```diff
@@ -521,27 +596,49 @@ async def new_session(
...
-        self._registry.create(session_id, log=log, cwd=str(cwd) if cwd else None)
-        logger.info(f"ACP NewSession: session_id={session_id}, cwd={cwd}")
+        # Register in the in-memory session registry (skip if already there)
+        if not self._registry.get(session_id):
+            self._registry.create(session_id, log=log, cwd=str(cwd) if cwd else None)
+        else:
+            # Touch existing session to update last_activity
+            existing = self._registry.get(session_id)
+            if existing:
+                existing.touch()
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/acp/agent.py]: Fixed in dcd7dff — now updates the log reference on existing registry entries.

File: gptme/acp/agent.py [hunk 2]
```diff
@@ -1016,14 +1125,42 @@ async def load_session(
...
+        modes = self._build_modes_state(session_id)
+        models = self._build_models_state(session_model)
+
+        # Schedule deferred notifications (commands, model info)
+        session = self._registry.get(session_id)
+        cwd = (session.cwd or "") if session else ""
+        if self._conn:
+            asyncio.create_task(
+                self._send_session_open_notifications(session_id, session_model, cwd)
+            )
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/acp/agent.py:1161]: Fixed in 7f35875 — load_session now passes resumed=True.

File: gptme/acp/agent.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/acp/agent.py]: This case is actually handled — see lines 637-639 in the `else` branch:

```python
if not resumed and log:
    existing_session.log = log
existing_session.touch()
```

When a registry entry exists but `log=None` (the scenario you're describing), `resumed` will be `False` (since we didn't find an existing log), so `not resumed and log` is `True`, and we update `existing_session.log = log`. The comment in the code even calls this out explicitly: *"Update log reference if we created a new one (e.g. registry entry existed with log=None)"*.

File: gptme/acp/agent.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/acp/agent.py:1161]: This is actually in the `load_session` path, where the session is *always* resumed (that's definitionally what `load_session` does). So `resumed=True` is the correct hardcoded value here — it's not a missing parameter, it's an intentional constant.

**Comment by TimeToBuildBob** [comment]
Rebased onto latest master. The CI fork-PR skip guard from #1441 is now properly preserved (was accidentally dropped in the consolidation). Clean diff now shows only new changes:

- `test.yml`: `continue-on-error` + `experimental` matrix for GLM-5
- `agent.py`: ACP improvements (agentInfo, session resume, code blocks, cleanup)
- `llm_openai.py`: OpenRouter `max_tokens` cap (prevents 402 credit errors)
- `test_acp_agent.py`: new test file

CI running.

File: gptme/acp/agent.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/acp/agent.py]: Addressed more completely in 9e1d088. The previous fix (`if not resumed and log`) correctly handled new sessions but still missed the case where `resumed=True` AND `existing_session.log is None` (e.g. session in registry from a prior partial init, but logfile exists on disk). Changed to `if log and existing_session.log is None` — this covers all cases, and added a test that explicitly exercises the disk-resume-with-null-log path.

**Comment by TimeToBuildBob** [comment]
Follow-up to Greptile review: added a more complete fix in 9e1d088.

The previous `dcd7dff` fix (`if not resumed and log`) correctly handled new sessions where the registry had `log=None`, but missed the case where `resumed=True` AND `existing_session.log is None` — i.e., a session existed in the registry without a log, but a conversation file was present on disk (so we load it and set `resumed=True`).

Changed the condition to `if log and existing_session.log is None` which covers all cases regardless of `resumed` state. Added a test (`test_registry_log_updated_when_resumed_from_disk_with_null_log`) that explicitly exercises this path.

**Comment by TimeToBuildBob** [comment]
CI status: all checks passing except GLM-5 (billing issue, not a code problem). The GLM failure is addressed by PR #1454 which marks it as experimental/continue-on-error. Once #1454 merges, this PR's GLM failure won't block merge.

**Comment by TimeToBuildBob** [comment]
PR #1454 has now merged. Updated this PR to restore test.yml to match master (with `continue-on-error` + `experimental: true` for GLM-5). 

CI status:
- ✅ All non-GLM checks passing
- ⚠️ GLM-5 check: fails due to billing (expected, `continue-on-error: true`)
- ✅ Overall workflow run: success
- ✅ PR is **MERGEABLE** (mergeStateStatus: UNSTABLE, no required checks blocked)

The PR now contains only ACP changes: `gptme/acp/agent.py` and `tests/test_acp_agent.py`.

**Comment by TimeToBuildBob** [comment]
@greptileai review
